src/js_native_api_v8.h

```diff
@@ -212,9 +212,12 @@ inline napi_status napi_set_last_error(napi_env env,
 #define NAPI_PREAMBLE(env)                                                     \
   CHECK_ENV((env));                                                            \
   RETURN_STATUS_IF_FALSE(                                                      \
-      (env),                                                                   \
-      (env)->last_exception.IsEmpty() && (env)->can_call_into_js(),            \
-      napi_pending_exception);                                                 \
+      (env), (env)->last_exception.IsEmpty(), napi_pending_exception);         \
+  RETURN_STATUS_IF_FALSE((env),                                                \
+                         (env)->can_call_into_js(),                            \
+                         (env->module_api_version == NAPI_VERSION_EXPERIMENTAL \
```

Comment:
Align this with how we decide to address https://github.com/nodejs/node/pull/47975/files#r1192497844.